presto-native-execution/presto_cpp/main/PrestoServer.h

```diff
@@ -172,6 +173,11 @@ class PrestoServer {
   /// Invoked to enable stats reporting and register counters.
   virtual void enableWorkerStatsReporting();
 
+  /// Invoked to initialize Presto to Velox plan validator.
+  virtual void initPrestoToVeloxPlanValidator();
```

Comment:
better to turn this into a factory function so users don't need to learn that it actually has side-effects (it sets the class member, can't be called twice, etc). 
```
virtual std::shared_ptr<VeloxPlanValidator> makePlanValidator();
```

and then callers can use this with more flexibility:

```
validator_ = makePlanValidator();
or
TaskResource(makePlanValidator(), ...);
```

Reply:
Could also drop the "PrestoTo" from the name since this, in effect, only validates a Velox query fragment.